libtenzir/builtins/aspects/partitions.cpp

```diff
@@ -76,6 +76,14 @@ class plugin final : public virtual aspect_plugin {
       event.field("schema").data(synopsis.synopsis->schema.name());
       event.field("schema_id")
         .data(synopsis.synopsis->schema.make_fingerprint());
+      auto add_resource = [&](std::string_view key, const resource& value) {
+        auto x = event.field(key).record();
+        x.field("url").data(value.url);
```

Comment:
A `URI` is an _identifier_ but not necessarily _addressable_ by itself. In this context, we really need to have a location.

Reply:
The main reason why I made the suggestion is because we also talk about URIs in the context of `from` and `to`, and I'd prefer having a consistent language here that's not confusing to the user. URI isn't wrong, it's just not as precise as URL in this context.

Reply:
> ideally stores and indexes are just loaded and stored using these operators (or some variant of them.

I fully agree on that part.

> it really depends on whether we have plans to support some form of URI that is not a URL with these operators in the future.

I'm not aware of such plans. Wouldn't we require a function to translate from a non-URL to a URL to make it usable.

In general I slightly lean towards URL, because it clearly conveys the semantics in the docs.

@eliaskosunen do you have an opinion on this?

Reply:
Hmh, I guess URL really would be the more accurate name for whatever we're doing: we're using the URI schema as a protocol, and reading from/writing to the location addressed by it. URI isn't the wrong term to use here, but URL is indeed more accurate. I can't really see us supporting non-URL URIs (what would it mean to read from a URN?).

I wouldn't be against doing a change for URI -> URL, but that'd need to be done everywhere.

Reply:
We've discussed this internally and Elias will open a new PR that changes from URI to URL where necessary. I think this PR is done then.